### PR TITLE
Add competition metrics

### PR DESCRIPTION
**Pull request by trushkin-ilya**

### CODE REVIEW

**Comment by trushkin-ilya** [comment]
@grib0ed0v review the changes please.

File: test.py [hunk 1]
```diff
@@ -1,48 +1,50 @@
...
+def evaluate(dataloader: data.DataLoader, model: nn.Module):
+    device = next(model.parameters()).device
     model.eval()
     print("Evaluating...")
-    val_loss = []
-    val_acc = []
+    tp, tn, fp, fn = 0, 0, 0, 0
     with torch.no_grad():
-        for i, batch in enumerate(dataloader):
+        for i, batch in enumerate(tqdm(dataloader)):
```
Comment (grib0ed0v) [inline test.py:20]: add tqdm to requirements.txt

Comment (trushkin-ilya) [reply test.py:20]: `tqdm` is already there.

File: datasets.py [hunk 2]
```diff
@@ -69,7 +68,9 @@ def __getitem__(self, idx):
         label = self.items[idx][1]
         img_path = os.path.join(self.dir, img_name)
         img = Image.open(img_path)
-        img = self.transform(img)
+        if self.transform:
+            img = self.transform(img)
+        img = transforms.ToTensor()(img)
```
Comment (grib0ed0v) [inline datasets.py]: It's better to use ToTensor() transform as a part of self.transform(img) call. Just construct more complex transform with transform.Compose()

File: submit.py [hunk 3]
```diff
@@ -45,18 +45,20 @@
...
                     inputs = inputs.to(device)
                     outputs = model(inputs)
-                    liveness_prob = F.softmax(outputs, dim=1)[0][1]
-                    video_id = dataset.get_video_id(i)
-                    if video_id not in result:
-                        result[video_id] = []
-                    result[video_id].append(liveness_prob)
-            for video_id, frame_probs in result.items():
-                submission.write(f'{video_id} {np.mean(frame_probs):.5f}\n')
+                    liveness_prob = F.softmax(outputs, dim=1)[0][1].item()
```
Comment (grib0ed0v) [inline submit.py]: Are you sure that you need to double subscript indexes [0][1] here?

File: test.py [hunk 4]
```diff
@@ -1,48 +1,50 @@
+from utils import SplittedDataLoader
```
Comment (grib0ed0v) [inline test.py]: it's better to group statements `from smt import SMT` and `import something`. And it's better to start file with import statements.

File: test.py [hunk 5]
```diff
@@ -1,48 +1,50 @@
...
-    dataloader = data.DataLoader(dataset, batch_size=args.batch_size)
+    dataset = CasiaSurfDataset(args.protocol, dir=args.data_dir, transform=transforms.Resize((320, 240)))
+    dataloader = SplittedDataLoader(dataset, train_batch_size=1, val_batch_size=args.batch_size)
     model = models.mobilenet_v2(num_classes=args.num_classes)
-    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
-    model.load_state_dict(torch.load(args.checkpoint, map_location=device))
-    loss_fn = nn.CrossEntropyLoss()
-    avg_loss, avg_acc = evaluate(dataloader, model, loss_fn)
-    print(f'Average loss: {avg_loss}, Average Accuracy: {avg_acc}')
+    model.load_state_dict(torch.load(args.checkpoint, map_location='cpu'))
```
Comment (grib0ed0v) [inline test.py]: It would be better to create variable `device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')` and use it here.

**Comment by trushkin-ilya** [comment]
@grib0ed0v review ASAP please

**Review by grib0ed0v — APPROVED** [review]

**Pull request merged**